lib/src/printers/prefix_printer.dart

```diff
@@ -0,0 +1,29 @@
+
+import 'package:logger/src/logger.dart';
+import 'package:logger/src/log_printer.dart';
+
+
+/// A decorator that can prefix any log based on the log level. Supply custom strings
+/// for the prefix for each level.
+class PrefixPrinter extends LogPrinter {
+  final LogPrinter _realPrinter;
+  Map<Level, String> _prefixMap;
+
+  PrefixPrinter(this._realPrinter,
+      {debug, verbose, wtf, info, warning, error, nothing}) : super() {
+    _prefixMap = {
+      Level.debug: debug ?? '  DEBUG ',
```

Comment:
What's with the leading and trailing white space? At first I thought it was to align text but the strings vary in length.

Reply:
Yea the idea was to line them up (not dart-savvy enough to easily do string formatting and was lazy) - I suppose `Level.NOTHING` doesn't line up but I guess even specifying it in the first place is redundant. Smarter fix is probably to dynamically set the padded length of the string to be the max of them all. That way if you pass in a custom message to the constructor you don't have to make sure it lines up with everything else. I should update that.

Reply:
Updated this to dynamically pad strings to the length of the longest prefix.